src/libraries/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -487,12 +487,42 @@ private void GetRgbValues(out int r, out int g, out int b)
             b = (int)(value & ARGBBlueMask) >> ARGBBlueShift;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void MinMaxRgb(out int min, out int max, int r, int g, int b)
+        {
+            if (b < g)
+            {
+                if (b < r)
+                {
+                    min = b;
+                    max = r < g ? g : r;
+                }
+                else
+                {
+                    min = r;
+                    max = g;
+                }
+            }
+            else
+            {
+                if (r < b)
+                {
+                    max = b;
+                    min = g < r ? g : r;
+                }
+                else
+                {
+                    max = r;
+                    min = g;
+                }
+            }
+        }
```

Comment:
This can be simplified to instead be:
```suggestion
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        private static void MinMaxRgb(out int min, out int max, int r, int g, int b)
        {
            if (r > g)
            {
                max = r;
                min = g;
            }
            else
            {
                max = g;
                min = r;
            }

            if (b > max)
            {
                max = b;
            }
            else if (b < min)
            {
                min = b;
            }
        }
```

Reply:
Thanks @stephentoub , I ran the perf_color microbenchmarks against 1) before this PR changeset, 2) the current PR's changeset, 3) proposed simplification

<details> 
  <summary>**.NET6 Before VS .NET6 with current PR changeset: Run1**</summary>
summary:
better: 3, geomean: 1.151
total diff: 3

No Slower results for the provided threshold = 2% and noise filter = 25ns.

| Faster                                        | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
| --------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Drawing.Tests.Perf_Color.GetBrightness |      1.23 |          1549.96 |          1259.50 |         |
| System.Drawing.Tests.Perf_Color.GetHue        |      1.18 |          1753.24 |          1484.57 |         |
| System.Drawing.Tests.Perf_Color.GetSaturation |      1.05 |          1451.59 |          1383.79 |         |
</details>

<details> 
  <summary>**.NET6 Before VS .NET6 with current PR changeset: Run2**</summary>
summary:
better: 3, geomean: 1.208
total diff: 3

No Slower results for the provided threshold = 2% and noise filter = 25ns.

| Faster                                        | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
| --------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Drawing.Tests.Perf_Color.GetBrightness |      1.23 |          1543.19 |          1259.71 |         |
| System.Drawing.Tests.Perf_Color.GetSaturation |      1.22 |          1682.03 |          1384.17 |         |
| System.Drawing.Tests.Perf_Color.GetHue        |      1.18 |          1758.68 |          1485.53 |         |
</details>

<details> 
  <summary>**.NET6 Before VS .NET6 with current PR changeset: Run3**</summary>
summary:
better: 3, geomean: 1.189
total diff: 3

No Slower results for the provided threshold = 2% and noise filter = 25ns.

| Faster                                        | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
| --------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Drawing.Tests.Perf_Color.GetSaturation |      1.22 |          1682.42 |          1384.03 |         |
| System.Drawing.Tests.Perf_Color.GetBrightness |      1.19 |          1497.18 |          1259.28 |         |
| System.Drawing.Tests.Perf_Color.GetHue        |      1.16 |          1725.25 |          1483.41 |         |
</details>

----------------------
----------------------

<details> 
  <summary>**.NET6 Before VS .NET6 with proposed simplification: Run1**</summary>
summary:
better: 2, geomean: 1.123
total diff: 2

No Slower results for the provided threshold = 2% and noise filter = 25ns.

| Faster                                        | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
| --------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Drawing.Tests.Perf_Color.GetBrightness |      1.13 |          1549.96 |          1375.52 |         |
| System.Drawing.Tests.Perf_Color.GetHue        |      1.12 |          1753.24 |          1565.75 |         |
</details>

<details> 
  <summary>**.NET6 Before VS .NET6 with proposed simplification: Run2**</summary>
summary:
better: 3, geomean: 1.135
total diff: 3

No Slower results for the provided threshold = 2% and noise filter = 25ns.

| Faster                                        | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
| --------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Drawing.Tests.Perf_Color.GetSaturation |      1.16 |          1682.03 |          1448.45 |         |
| System.Drawing.Tests.Perf_Color.GetHue        |      1.12 |          1758.68 |          1566.17 |         |
| System.Drawing.Tests.Perf_Color.GetBrightness |      1.12 |          1543.19 |          1375.17 |         |
</details>

<details> 
  <summary>**.NET6 Before VS .NET6 with proposed simplification: Run3**</summary>
summary:
better: 2, geomean: 1.132
worse: 1, geomean: 1.050
total diff: 3

| Slower                                        | diff/base | Base Median (ns) | Diff Median (ns) | Modality|
| --------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Drawing.Tests.Perf_Color.GetBrightness |      1.05 |          1497.18 |          1572.36 |         |

| Faster                                        | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
| --------------------------------------------- | ---------:| ----------------:| ----------------:| --------:|
| System.Drawing.Tests.Perf_Color.GetSaturation |      1.16 |          1682.42 |          1446.59 |         |
| System.Drawing.Tests.Perf_Color.GetHue        |      1.10 |          1725.25 |          1565.42 |         |
</details>
----------------------
----------------------

Some of the benchmarks seem to get a bit slower on average after the simplification. I think what is happening may be two causes:
1) There are paths with an extra store (we store max & min in the first conditional and then if the second conditional is taken one of these is overwritten).
2) Just my guess, but processor pipelining with branch prediction may find it easier to go through each of the conditionals in the current implementation as there are no dependencies. The simplification introduces a dependency on the second conditional as it needs the values from the first? Just a guess here.

I will follow yours and @safern 's recommendation on which to keep. Thanks again.

Reply:
I didn't look at the disassembly but there might be an aliasing complication also, if the JIT compiler has to account for the possibility that `min` and `max` refer to the same location. Although perhaps it can disprove that after inlining.

Reply:
I actually see them get faster on my machine.  Regardless, though, I'd rather keep the code smaller, especially with it triplicated due to the aggressive inlining.

Reply:
One more possible improvement - using local valiables instead of ```out``` parameters:

```csharp
		private static void MinMaxRgb(out int min, out int max, int r, int g, int b)
		{
			int minLoc, maxLoc;
			if (r > g)
			{
				maxLoc = r;
				minLoc = g;
			}
			else
			{
				maxLoc = g;
				minLoc = r;
			}

			if (b > maxLoc)
			{
				maxLoc = b;
			}
			else if (b < minLoc)
			{
				minLoc = b;
			}

			max = maxLoc;
			min = minLoc;
		}
```

Reply:
> I actually see them get faster on my machine. Regardless, though, I'd rather keep the code smaller, especially with it triplicated due to the aggressive inlining.

Sounds good @stephentoub , I've updated it with the requested changes. (I'm not sure how crediting works in this project as this is my second pull request, so I've added your username to the commit details as the originator of the simplification).

> One more possible improvement - using local valiables instead of `out` parameters:
Thanks @hypeartist, I'm wondering if these local variables will be eliminated due to inlining, so performance will be about the same?

Reply:
> I'm not sure how crediting works in this project as this is my second pull request, so I've added your username to the commit details as the originator of the simplification

Thanks, but not necessary... I comment on _a lot_ of PRs :)

Reply:
> I'm wondering if these local variables will be eliminated due to inlining, so performance will be about the same?

Yes, e.g.
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdvHgwCy5ABRWMDKGgZOGAczcfgASmNTc2CbKxtsBDZPYAcIDmcPfCs3OITrfAi3Vy83fw1qYODiAHYGDIQGODKrfOCAXyCzRot3axtSR2tsj29W539AgsKzUK5wyOiO1L7qrhT4mfKst1685qLS8srZ2vMGoaaDhgBtGxgMAAsIABMASXFJezPLm/uxSQB5MT4ILlwWACCnk8sFwuF4Chgty4kisVk8fgAus0xFAIdgMDAmOQkEwULYwhEojF7NNEskGGT0pkZt1rL1fH5miYjsFeAAzBj2KAMAB8XiZrPMLOGwy2AF4XLtReYklwGJLPNLCvsZTBJLgYOthdqxREFV5lTK5QaoEa9rr3Jz7MA+WUIoKZSKZbL9ZLgOazKrRerNVaubaADyzR2i50u2YGj2W71e5rNU7nK53B5PJOvB5fH5/QHA0HgyHQ2FceFIlFohQYrHMXHEfGjcbEqYLclzSkt6kIHxdbvOBnWfzMy2tgAyEDAbnKY7Anv93LtCMt4ZdU/HptnwTl04NSpjlt9WqFZmXxoi28VG9lVnPUr31GH1tt/NXYFDwxPopfUdnscKB7nQazNOb6FB+YrXmu7o/poR72hUkovrOJoIRBM7NPsdRAA===